### ISSUE TITLE
no-role-prefix do not work in role

### ISSUE THREAD

**Issue by nesies** [ansible/ansible-lint#3668]
##### Summary

var-naming[no-role-prefix] is not detected when calling a role in "role" section

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```
$ python -V
Python 3.9.16
```

```console (paste below)
$ ansible-lint --version
ansible-lint 6.17.3.dev20 using ansible-core:2.15.1 ansible-compat:4.1.2 ruamel-yaml:None ruamel-yaml-clib:None
```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```console (paste below)
mkdir -p ansible_collections/test/role_prefix/roles/test_role1
mkdir -p ansible_collections/test/role_prefix/roles/test_role2

cat > playbook_test_role_prefix.yaml << EOF
---
- name: Test role-prefix
  hosts: localhost
  connection: local
  roles:
    # is ko but not detected
    - role: test.role_prefix.test_role1
      var1: val1
    # is ok
    - role: test.role_prefix.test_role2
      test_role2_var2: val2
  tasks:
    # is ko and detected
    - name: Include1
      ansible.builtin.include_role:
        name: test.role_prefix.test_role1
      vars:
        var1: val1
    # is ok
    - name: Include2
      ansible.builtin.include_role:
        name: test.role_prefix.test_role2
      vars:
        test_role2_var2: val2
EOF

export ANSIBLE_COLLECTIONS_PATH=$PWD
```

run
```
$ ansible-lint --offline playbook_test_role_prefix.yaml
WARNING  Listing 1 violation(s) that are fatal
var-naming[no-role-prefix]: Variables names from within roles should use role_name_ as a prefix. (vars: var1)
playbook_test_role_prefix.yaml:18 Task/Handler: Include1

Read documentation for instructions on how to ignore specific rule violations.

                    Rule Violation Summary
 count tag                        profile rule associated tags
     1 var-naming[no-role-prefix] basic   idiom
```

##### Desired Behavior

var-naming[no-role-prefix] violation should be detected in the first and third role call

first:
```
    # is ko but not detected
    - role: test.role_prefix.test_role1
      var1: val1
```
third: case
```
    # is ko and detected
    - name: Include1
      ansible.builtin.include_role:
        name: test.role_prefix.test_role1
      vars:
        var1: val1
```

##### Actual Behavior

var-naming[no-role-prefix]  should detect first and third calls

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
